### PR TITLE
f32 docs: define 'arithmetic' operations

### LINKED ISSUE #129699: Float docs do not define the set of "arithmetic" operations
Pointed out by @workingjubilee in https://github.com/rust-lang/rust/pull/129683: the float docs are using the term "arithmetic" operations in a few places, but that term is never defined. Furthermore, the new NaN docs probably should use that term as well when discussing which operations the NaN rules apply to (instead of "non-bitwise floating point operations" which they use currently).

By "arithmetic operations", the docs likely mean all operations except for unary `-`, `abs`, `copysign`, `signum`, `{to,from}_bits`, `{to,from}_{be,le,ne}_bytes` and `is_sign_{positive,negative}`.

### PR DESCRIPTION
**Pull request by RalfJung**
r? @workingjubilee 
Fixes https://github.com/rust-lang/rust/issues/129699

### CODE REVIEW

File: library/core/src/primitive_docs.rs
Comment (workingjubilee) [inline library/core/src/primitive_docs.rs:1240]: Optional: list format?
```suggestion
/// The following rules apply when a NaN value is returned from an arithmetic operation:
/// - The result has a non-deterministic sign.
/// - The quiet bit and payload are non-deterministically chosen from
///    the following set of options:
```

File: library/core/src/primitive_docs.rs
Comment (workingjubilee) [inline library/core/src/primitive_docs.rs:1279]: The hyphen is actually correct. Consider encountering:
- a friendly-looking logician, versus
- a friendly little logician

Notice that "looking logician" doesn't hold up as a descriptor, whereas "little logician" does, and that's why they get hyphenated.
```suggestion
/// When an arithmetic floating-point operation is executed in `const` context, the same rules
```

Also consider the titles here:
- https://en.wikipedia.org/wiki/Floating-point_arithmetic
- https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html

File: library/core/src/primitive_docs.rs
Comment (workingjubilee) [inline library/core/src/primitive_docs.rs:1222]: Oh no I noticed the hyphens.
```suggestion
/// This section defines the possible NaN bit patterns returned by floating-point operations.
```

File: library/core/src/primitive_docs.rs
Comment (workingjubilee) [inline library/core/src/primitive_docs.rs:1224]:
```suggestion
/// The bit pattern of a floating-point NaN value is defined by:
```

File: library/core/src/primitive_docs.rs
Comment (workingjubilee) [reply library/core/src/primitive_docs.rs:1279]: But also this is actually probably supposed to be
```suggestion
/// When a floating-point arithmetic operation is executed in `const` context, the same rules
```

**Review by workingjubilee — APPROVED** [review]
Thanks!
r=me with the hyphenation nit addressed

File: library/core/src/primitive_docs.rs
Comment (RalfJung) [reply library/core/src/primitive_docs.rs:1240]: That makes the lists nested then, since there already is a list that defines the quiet bit and payload options. So the following items should all be indented. Do you think that's a good idea?

File: library/core/src/primitive_docs.rs
Comment (RalfJung) [reply library/core/src/primitive_docs.rs:1279]: I removed the hyphen since the rest of the file doesn't use a hyphen. I can also add it everywhere.

For the 2nd one, the order of adjectives in English follows arcane rules so I won't argue about it. ;)

Comment (RalfJung) [reply library/core/src/primitive_docs.rs:1279]: Hm actually no I think I want to argue. The emphasis here is on "arithmetic", so putting that first seems more clear to me. This distinction of "arithmetic" and "non-arithmetic" is also a floating-point thing, so it's not like we have an "arithmetic operation" in its floating-point variant.

So, "arithmetic floating-point operation" makes more sense to me.

**Comment by RalfJung** [comment]
Note that there are still >400 occurrences of "floating point" in other files in `library/`. I'll leave it to someone else to take care of them. :)

File: library/core/src/primitive_docs.rs
Comment (workingjubilee) [reply library/core/src/primitive_docs.rs]: Hm, okay!

File: library/core/src/primitive_docs.rs
Comment (workingjubilee) [reply library/core/src/primitive_docs.rs:1240]: Yeah, I think so? If it turns out to be gross when actually rendered then not listifying this is fine.

**Comment by workingjubilee** [comment]
To make things even more confusing, there are a few occasions where "floating point" would actually be correct... though it probably should be avoided in those cases (e.g. when used as a noun-phrase).

But yes I'll probably go hunt them down when I feel bored. :^)

File: library/core/src/primitive_docs.rs
Comment (RalfJung) [reply library/core/src/primitive_docs.rs]: Seems okay:

image

**Comment by workingjubilee** [comment]
cool!

@bors r+ rollup

**Pull request merged**
